var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -265,16 +271,27 @@ class Petsc(Package, CudaPackage, ROCmPackage):
-    depends_on('hypre@2.14:2.22.0+mpi~internal-superlu~int64', when='@3.14:3.15+hypre+mpi~complex~int64')
-    depends_on('hypre@2.14:2.22.0+mpi~internal-superlu+int64', when='@3.14:3.15+hypre+mpi~complex+int64')
-    depends_on('hypre@2.14:+mpi~internal-superlu~int64', when='@3.16:+hypre+mpi~complex~int64')
-    depends_on('hypre@2.14:+mpi~internal-superlu+int64', when='@3.16:+hypre+mpi~complex+int64')
-    depends_on('hypre@develop+mpi~internal-superlu+int64', when='@main+hypre+mpi~complex+int64')
-    depends_on('hypre@develop+mpi~internal-superlu~int64', when='@main+hypre+mpi~complex~int64')
+    depends_on('hypre@:2.13+mpi+fortran~internal-superlu~int64', when='@:3.8+hypre+mpi~complex~int64+fortran')
+    depends_on('hypre@:2.13+mpi+fortran~internal-superlu+int64', when='@:3.8+hypre+mpi~complex+int64+fortran')
+    depends_on('hypre@2.14:2.18.2+fortran+mpi~internal-superlu~int64', when='@3.9:3.13+hypre+mpi~complex~int64+fortran')
+    depends_on('hypre@2.14:2.18.2+fortran+mpi~internal-superlu+int64', when='@3.9:3.13+hypre+mpi~complex+int64+fortran')
+    depends_on('hypre@2.14:2.22.0+fortran+mpi~internal-superlu~int64', when='@3.14:3.15+hypre+mpi~complex~int64+fortran')
+    depends_on('hypre@2.14:2.22.0+fortran+mpi~internal-superlu+int64', when='@3.14:3.15+hypre+mpi~complex+int64+fortran')
+    depends_on('hypre@2.14:+fortran+mpi~internal-superlu~int64', when='@3.16:+hypre+mpi~complex~int64+fortran')
+    depends_on('hypre@2.14:+fortran+mpi~internal-superlu+int64', when='@3.16:+hypre+mpi~complex+int64+fortran')
+    depends_on('hypre@develop+fortran+mpi~internal-superlu+int64', when='@main+hypre+mpi~complex+int64+fortran')
+    depends_on('hypre@develop+fortran+mpi~internal-superlu~int64', when='@main+hypre+mpi~complex~int64+fortran')
+    # without fortran
+    depends_on('hypre@:2.13+mpi~fortran~internal-superlu~int64', when='@:3.8+hypre+mpi~complex~int64~fortran')
+    depends_on('hypre@:2.13+mpi~fortran~internal-superlu+int64', when='@:3.8+hypre+mpi~complex+int64~fortran')
+    depends_on('hypre@2.14:2.18.2~fortran+mpi~internal-superlu~int64', when='@3.9:3.13+hypre+mpi~complex~int64~fortran')
+    depends_on('hypre@2.14:2.18.2~fortran+mpi~internal-superlu+int64', when='@3.9:3.13+hypre+mpi~complex+int64~fortran')
+    depends_on('hypre@2.14:2.22.0~fortran+mpi~internal-superlu~int64', when='@3.14:3.15+hypre+mpi~complex~int64~fortran')
+    depends_on('hypre@2.14:2.22.0~fortran+mpi~internal-superlu+int64', when='@3.14:3.15+hypre+mpi~complex+int64~fortran')
+    depends_on('hypre@2.14:~fortran+mpi~internal-superlu~int64', when='@3.16:+hypre+mpi~complex~int64~fortran')
+    depends_on('hypre@2.14:~fortran+mpi~internal-superlu+int64', when='@3.16:+hypre+mpi~complex+int64~fortran')
+    depends_on('hypre@develop~fortran+mpi~internal-superlu+int64', when='@main+hypre+mpi~complex+int64~fortran')
+    depends_on('hypre@develop~fortran+mpi~internal-superlu~int64', when='@main+hypre+mpi~complex~int64~fortran')
```

Comment:
With the current code - does adding the following also work?

```
  depends_on('hypre+fortran, when='+hypre+fortran')
  depends_on('hypre~fortran, when='+hypre~fortran')
```

[perhaps some of the current code can be similarly simplified]

Reply:
Thanks for the suggestion! This seems to do the trick. Adopted this approach in the PR.